lessons/git/intro/practice-code-2017-10-05.sh

```diff
@@ -0,0 +1,168 @@
+## for Git commands, typing git <command> --help will show a help page
+
+## Configure your local git setup. This only needs to be done once. Only the first two are essential.
+git config --global user.name "Your Name"
+git config --global user.email "you@some.domain"
+
+## other optional configurations
+git config --global color.ui "auto" 
+
+## setup an editor for text files (if you need one)
+## you can use any editor you like, such as atom, notepad, vim, etc.
+## more help on setting up editors
+## https://help.github.com/articles/associating-text-editors-with-git/
+git config --global core.editor "your_editor"
+
+## List all the configurations completed
+git config --list
+
+## cd = change directory (move to another folder)
+## note: "cd .." moves you back one folder
+cd ~/Documents
+## for windows users with full path
+cd C:/Users/user_name/Documents
+
+## mkdir = make directory (create a folder)
+mkdir git_repos
+
+## move to the GitHub folder
+cd git_repos
+
+## clone = copy the repo from GitHub to your computer
+## In the terminal (still in the GitHub folder), type:
+git clone https://github.com/your-account-name/studyGroup
+cd studyGroup
+
+## creating a local git repo and pushing it to your origin on GitHub
+cd ..
+mkdir new_local_repo
+cd new_local_repo
+
+## convert this folder to a git repo
+git init
+
+
+## First, fork the 'studyGroup' repo from the UofTCoders group.  Forking
+## just means copy over into your own account. Click "fork" in the top right corner.
+## url: https://github.com/UofTCoders/studyGroup
+
+
+
+## Those with internet problems can do this: create a folder called
+## sandbox, move into that folder, and create an empty git repo (init)
+## git init = initialize a git repo (start an empty git repo in that folder)
+mkdir sandbox && cd sandbox && git init
+
+## ls = list files and folders in the current location
+ls
+
+## pwd = print working directory (see what your current location is)
+pwd
+
+## git status = check the status of your folder - see if there are any changes.
+git status
+
+## move into the _data folder, see what's there
+cd _data
+ls
+
+## cat = prints to the screen the contents of a file (also used to join files together)
+cat members.yml
+
+## Add yourself to the members list!
+## typing "nano filename" opens the file in the editor nano. 
+## you can use any editor you like, such as atom, notepad, vim, etc.
+nano members.yml
+
+## git status = check the status of your repository
+## you'll see a changed file
+git status
+
+## add = track and put the file into the git staging area (staging
+## just means ready to be saved into the history, but not actually
+## saved)
+git add members.yml
+git status ## just to see what happened
+
+## commit = save everything in the staging area into the history (the
+## git repo).  -m = the commit message to be included in the history.
+## Try to be descriptive with this message.
+git commit -m "Added myself to the members list"
```

Comment:
Consider changing `Added` to `Add`.

Although the recommendations on this differ widely, the imperative style is preferred by the git project itself, and it's the closest I have seen to an official guideline https://stackoverflow.com/questions/3580013/should-i-use-past-or-present-tense-in-git-commit-messages

Reply:
Not relevant to current version of file. Ignored.